fix: validate JSON parameter

Roming22 opened the pull request:
We do not expect any of the key/values in the dockerconfigjson to have spaces, and interpret the presence of a space as bad input.

This safeguard is added after a customer ran into a difficult to debug issue where the invalid content of the secret had an unexpected impact on the ability to pull any image in the namespace.

cf [RHTAP-5655](https://issues.redhat.com//browse/RHTAP-5655)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Added JSON validation for credential inputs, enforcing no spaces in keys or string values and returning clearer errors for invalid JSON.
  - Applied this validation to image registry configurations before URL checks for more reliable setup feedback.

- Bug Fixes
  - Prevents saving misconfigured image registry settings with malformed or space-containing JSON, reducing setup errors.

- Tests
  - Added comprehensive tests covering valid/invalid URLs and multiple JSON scenarios (syntax errors, spaces in keys/values).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dperaza4dustbit commented on pkg/integration/helper.go:
```diff
@@ -22,3 +23,46 @@ func ValidateURL(location string) error {
+}
+
+// ErrJSONContainsSpaces is an error returned when a JSON key or value contains spaces.
+var ErrJSONContainsSpaces = errors.New("contains unexpected spaces")
```
Kind of personal style thing but if you are OK, I would prefer all error definitions at the beginning of the file

Roming22 replied:
Done

dperaza4dustbit commented on pkg/integration/helper.go:
```diff
@@ -22,3 +23,46 @@ func ValidateURL(location string) error {
+		}
+	case map[string]interface{}:
+		for key, val := range v {
+			if strings.Contains(key, " ") {
```
You could potentially remove this check if you just call checkJSONSpaces on both key and value the check will be done on the next call.

Roming22 replied:
Done

dperaza4dustbit commented on pkg/integration/helper.go:
```diff
@@ -22,3 +23,46 @@ func ValidateURL(location string) error {
+	switch v := data.(type) {
+	case string:
+		if strings.Contains(v, " ") {
+			return fmt.Errorf("--%s %w in value %q", p, ErrJSONContainsSpaces, v)
```
you would probably say "string" instead of "value" if you decide to call recursion on both key and value if there is a map

Roming22 replied:
Done

Roming22 committed:
fix: validate JSON parameter

We do not expect any of the key/values in the dockerconfigjson to have
spaces, and interpret the presence of a space as bad input.

This safeguard is added after a customer ran into a difficult to debug
issue where the invalid content of the secret had an unexpected impact
on the ability to pull any image in the namespace.

cf RHTAP-5655

dperaza4dustbit reviewed (approved):
/lgtm

Roming22 commented:
/retest

Roming22 commented:
CI is good enough to force merge.

<img width="1082" height="379" alt="image" src="https://github.com/user-attachments/assets/6e5d3218-7222-41b4-baa8-c0e45e974358" />
